app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -66,7 +66,7 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
      * Used to cancel background->UI threads: we attach them as children to this job
      * and cancel this job at the end of the UI lifecycle, cancelling the children.
      */
-    private var uiLifecycleCancelJob: Job
+    var uiLifecycleCancelJob: Job
```

Comment:
You've got the code in place where it works – great! But is it maintainable?

In general, each abstraction layer should manage anything it allocates and shouldn't pass it to its children or parents.

In this specific case, should BrowserFragment be controlling the lifecycle of a member of the BrowserNavigationOverlay's? Instead, I think BrowserFragment should own the uiLifecycleCancelJob (I'll let you figure out what I mean here).

A few concrete examples why this is less maintainable:
- What if BrowserFragment has another child that requires a uiLifecycleCancelJob? What do we have to do to make that work?
- What if BrowserNavigationOverlay, a View, is re-used in another Fragment or Activity? Will we remember to reach inside it to cancel the lifecycle job?